enterprise/coderd/workspaces_test.go

```diff
@@ -1420,6 +1422,182 @@ func TestTemplateDoesNotAllowUserAutostop(t *testing.T) {
 	})
 }
 
+// TestWorkspaceTagsTerraform tests that a workspace can be created with tags.
+// This is an end-to-end-style test, meaning that we actually run the
+// real Terraform provisioner and validate that the workspace is created
+// successfully. The workspace itself does not specify any resources, and
+// this is fine.
+func TestWorkspaceTagsTerraform(t *testing.T) {
```

Comment:
Woah, he did it :exploding_head: 

This is a super nice test. Slightly concerned with terraform being executed in tests, would be nice if we could lock it down somehow to prevent it allocating any resources.

Reply:
I could add a hacky check for the pattern `resource "[^"]+" "[^"]+"` and fail. That would at least make folks think twice before hacking it to allow certain resources.

Reply:
I don't think ruleguard would be sufficient. I wonder if we can do something to the executable. Run it another userspace with no perms? Configure a `HTTP_PROXY` that is broken so any external requests fail? I'm not sure :man_shrugging: 

I'm ok if we let this slide for now, since I don't know a good way. Just an unfortunate consequence.

Reply:
Yeah I think we may need to think about the best way to prevent misuse of this.